# Instructions {#instructions}

## Goal of the exercise

The goal is to generate high quality metagenome assembled genomes (MAGs), identify the taxonomy, and the metabolic potential of the MAGs of interest.

You will be working with metagenomic data from Cruise \#72. You will be provided with prerequisite information, such as a merged profile database and contigs database, that you would use to bin MAGs and later classify the taxonomy. You would be working on an 'omics analysis and visualization platform, Anvi'o. For binning, you would use MetaBAT2, MaxBin2, CONCOCT, and DAS Tool. For taxonomic classification you would use GTDB-Tk and finally to check genome completeness and redundancy you would use CheckM.

## Project ideas

-   Determine what fractions of contigs map to MAGs vs contigs not in MAGs, as a way to evaluate the total amount of information available to evaluate
-   Evaluate nitrogen cycling genes within MAGs using Prokka and TreeSAPP annotations -Compare taxonomic assignment TreeSAPP identified genes to GTDB-Tk assigned taxonomies of MAGs, generated using Anvi'o. Can you identify known groups based on published work -Evaluate changes in nitrogen cycling gene taxonomy and abundance at different depths in the Saanich inlet water column

## Reports and Assessment

Reports should be formatted as per the [Instructions to Authors](https://jb.asm.org/sites/default/files/additional-assets/JB-ITA.pdf) for the [Journal of Bacteriology](https://jb.asm.org/).

Reports will account for 20% of the course marks. The project will be assessed by the following criteria:

1.  Completion of the proposed workflow. Expected outputs are available and formatted correctly (40%).
2.  A written report (4000 - 4500 words) adhering to the structure below (40-80%). 3. Oral presentation following the structure of the written report with emphasis on visual analysis of results (20%).

Each group will complete **one** report with the sections listed below. The word counts for each section are recommnedations and can be adjusted by +/- 200 words, but the topics within each must be discussed!

### Abstract (250 word max.)

-   Concisely summarize the major results and implications of the report.

### Importance (120 word max.)

-   Lay explanation of the significance of the research performed to the field of environmental microbiology.

### Introduction (1250 words)

-   Introduce Saanich Inlet as a model ecosystem for studying microbial community responses to ocean deoxygenation e.g. seasonal cycles, relevant biogeochemistry, previous studies, etc.
-   Provide relevant information regarding metagenomic approaches (data generation and application), pros, cons, and alternatives.
-   Introduce one or all of: metagenomic binning, genome assembly, taxonomic profiling.
-   Justify reasoning for using cultivation-independent methods.

### Methods (750 words)

-   Explain the workflow to generate inputs for the project to show you understand each step.
-   Explain the tools used at each step, and why they were chosen, to answer your group's specific question.
-   Reference the bash and R(md) scripts containing all commands. Do not include the scripts themselves in the text of your manuscript but submit them as individual files on Canvas.
-   Explain why each step was used, what the outputs are, etc. For example, metagenome assembly was used to build larger sequences (contigs) from the sequencing reads to reduce data size and redundancy, as well as inputs for binning. **Don't be silly and plagiarise this**.
-   Explain any potential deviations from the default parameters or the suggested workflow.

### Results (1000 words)

-   Provide a summary of the CheckM, and GTDB-Tk outputs, as well as those from other software, if applicable.
-   Report on the Prokka annotations such as rRNA predicted, \# genes annotated, and potential metabolisms of the medium- and high-quality bins.
-   Report of the relevant results from your group's selected analysis.

### Discussion (1000 words)

-   Relate the biological results back to the introduction and what is or is not sensible based on the geochemical state of the environment. How do the different population genome bins in your data set contribute to the metabolic pathways you investigated? (e.g. denitrification, dissimilatory sulfate reduction and oxidation). Do they encode partial or complete pathways (i.e. do your data support a distributed pathway)?
-   Discuss your reasoning behind the steps you took to answer your group's specific question. - Additional questions worth pursuing further.
-   Briefly state any challenges encountered (we know there will be some!) and troubleshooting (parameter or software changes).

### Figures and tables (â‰¥4 with captions)

Some recommended (not required!) figures to include:

-   Geochemical gradients for nutrients (phosphate, silicate), nitrogen compounds (nitrate, nitrite, ammonia, nitrous oxide), oxygen and/or sulfur compounds. These are depth-dependent.
-   Scatter plot comparing contamination % (Y-axis) versus completion % (X-axis) between population genome bins. Consider colour coding the bins by taxonomic rank e.g. phylum, order, etc. Indicate high, medium and low contamination bins (see Canvas for an example).
-   RPKM bubble-plot of genes in a pathway versus taxonomy.

### References (\>20)

Necessary citations include:

-   Each software used (including those in the provided bash scripts (MEGAHIT, MetaBAT, CheckM, etc.).
-   Saanich Inlet data papers.

### Fin.

**This evaluation rubric should serve as a useful scaffold for your group. How you expand on it will depend on your combined interests and analytic choices. Let the science take you somewhere!**

### UJEMI submission

Outstanding reports will be invited to submit to The Undergraduate Journal of Experimental Microbiology and Immunology ([UJEMI](https://jemi.microbiology.ubc.ca/)). More information on this will be provided near the end of the Finals Period.
