# Instructions {#instructions}

## Goal of the exercise

The primary objective of this tutorial is to use anvi'o to construct population genome bins, also known as  metagenome assembled genomes (MAGs), to identify the taxonomy and metabolic potential of medium to high quality MAGs of interest [Bowers et al. 2017](https://rdcu.be/b9JqO). 

You will be working with metagenomic data sets from the Saanich Inlet time series Cruise \#72. Prerequisite input files are provided to construct individual contig and merged profile databases for binning. As indicated above you will be using anvi'o, an 'omics analysis and visualization platform' developed by the Meren lab at the University of Chicago. Specific automated binning applications invoked by anvi'o can include MetaBAT2, MaxBin2, CONCOCT, and DAS Tool. 

Following MAG construciton you will use CheckM to check MAG completeness and contamination and GTDB-Tk to assign taxonomic affiliation. Collectively this information will enable you to reconstruct metabolic networks at the individual, population and community levels of biological organization. You will also be using prokka to predict and annotate open reading frames, and BWA to map reads onto MAGs to determine coverage and assess how representative the bins in the sampled community. 

Results from anvi'o will be compared to those generated using TreeSAPP, an application for fast phylogentic mapping of functional anchor genes, and the subject of a separate tutorial.

## Project ideas

Note that these are guiding ideas only. You are welcome to follow your own interests in developing a cogent reporting plan. You are working with primary scientific data and are encouraged to let the science "take you somewhere".

-   Determine what fraction of reads map to MAGs versus contigs not in MAGs. Are medium to high quality MAGs more or less abundant in the sample based on read mapping?
-   Identify nitrogen cycling genes encoded in MAGs using prokka annotations and compare these results to TreeSAPP annotations using both MAGs and contigs. Are the results equivalent or different? 
-   Describe GTDB-Tk assigned taxonomies of MAGs. Can you identify previously described microbial groups based on the published record? 
-   Evaluate changes in nitrogen cycling gene taxonomy and abundance at different depths in the Saanich inlet water column with special emphasis on the modular denitrification pathway. How does the frequency distribution of denitrification genes change in relation to oxygen concentration and are there any gaps in pathway representation that are more likely to be filled by a specific taxonoic group?

## Reports and Assessment

Reports should be formatted as per the [Instructions to Authors](https://jb.asm.org/sites/default/files/additional-assets/JB-ITA.pdf) for the [Journal of Bacteriology](https://jb.asm.org/). These are roughly equivalent to the JEMI+ reporting guidelines (https://jemi.microbiology.ubc.ca/instructions/submission-guidelines).

Reports will account for 20% of the course marks (50 points). The project will be assessed by the following criteria:

1. Completion of the proposed workflow. Expected outputs are available and formatted correctly (40%).
2. A written report (4000 - 4500 words) adhering to the structure below (40%). 
3. Five minute oral presentation following the structure of the written report with emphasis on visual analysis of results (20%).

Each group will complete **one** report with the sections listed below. The word counts for each section are recommnedations and can be adjusted as needed, but the topics within each must needs be included!

### Abstract (250 word max.)

-   Concisely summarize the major results and implications of the report.

### Importance (120 word max.)

-   Lay explanation of the significance of the research performed to the field of environmental microbiology.

### Introduction (1250 words)

-   Introduce Saanich Inlet as a model ecosystem for studying microbial community responses to ocean deoxygenation e.g. seasonal cycles, relevant biogeochemistry, nitrogen cycle, relationship to oxygen minimum zones, previous studies, etc.
-   Provide relevant information regarding metagenomic approaches (data generation and application), pros, cons, and alternatives.
-   Introduce one or all of: metagenomic binning, genome assembly, taxonomic profiling.
-   Justify reasoning for using cultivation-independent methods to study microbial metabolic networks e.g. nitrogen cycle.
-   Here we describe "X, Y, Z". End this section by telling the reader what you are going to tell them. 

### Methods (750 words)

-   Explain the workflow to generate inputs for the project to show you understand each step.
-   Explain the tools used at each step, and why they were chosen, to answer your group's specific question.
-   Reference the bash and R(md) scripts containing all commands. Do not include the scripts themselves in the text of your manuscript but submit them as individual files on Canvas.
-   Explain why each step was used, what the outputs are, etc. For example, metagenome assembly was used to build contigs from the quality controlled sequencing reads and used as inputs for binning. **Don't be silly and plagiarise this**.
-   Explain any potential deviations from the default parameters or the suggested workflow.

### Results (1000 words)

-   Provide a summary of the binning results, CheckM, and GTDB-Tk outputs, as well as those from other software used, if applicable. Explain rationale for focusing on high-quality MAGs but provide information on the total number of bins constructed and their relative abundance based on read mapping.
-   Report on the Prokka annotations such as rRNA predicted, \# genes annotated, and potential metabolisms encoded in the medium to high-quality MAGs.
-   Describe the relevant results from your group's selected project ideas. Explain your experimental logic for each step in the analysis workflow.

### Discussion (1000 words)

-   Relate your results back to the introduction. Think about the environmental context and how the obseved taxonmy and function  are related to measured geochemical conditions in the Saanich Inlet water column. Consider how each MAG contributes to different metabolic interactions (e.g. denitrification, sulfur oxidation, amino acid biosynthesis, etc). Can you find evidence for distributed metabolism?
-   How do your results compare to previous studies? Are you able to make new observations that expand on prior knowledge? 
-   Briefly state any challenges encountered (we know there will be some!) and troubleshooting (parameter or software changes). Do new questions arise from your analysis? If you were to continue working on this proejct what future directions might you take to expand on the work or test specific hypotheses?

### Figures and tables (â‰¥4 with captions)

Some recommended figures to include:

-   Geochemical gradients for nutrients (phosphate, silicate), nitrogen compounds (nitrate, nitrite, ammonia, nitrous oxide), oxygen and/or sulfur compounds. These are depth-dependent.
-   Scatter plot comparing contamination % (Y-axis) versus completion % (X-axis) between population genome bins. Consider colour coding the bins by taxonomic rank e.g. phylum, order, etc. Indicate high, medium and low contamination bins (see Canvas for an example).
-   RPKM bubble-plot of genes in a pathway versus taxonomy.
-   A diagram showing the taxonomic identity and abundance for each step in the denitrification pathway.
-   A panel of phylogenetic trees showing the depth specific trends for selected TreeSAPP reference packages. 

### References (\>20)

Necessary citations include:

-   Each software application used (including those in the provided bash scripts (MEGAHIT, MetaBAT, CheckM, etc.). Make sure you include version numbers in your methods and indicate parameter settings used e.g. default. 
-   Relevant papers from Saanich Inlet or other marine oxygen minimum zones.

### Fin.

**This evaluation rubric should serve as a useful scaffold for your group. How you expand on it will depend on your combined interests and analytic choices. Let the science take you somewhere!**

### UJEMI submission

Outstanding reports will be invited to submit to The Undergraduate Journal of Experimental Microbiology and Immunology ([UJEMI](https://jemi.microbiology.ubc.ca/)).
