# Introduction to anvi'o

## What is anvi'o

Analysis and visualization platform for 'omics data, or anvi'o was developed by [Eren et al. 2015](https://peerj.com/articles/1319/) at the University of Chicago. Anvi'o is an open source platform for analyzing integrated multi-omics data and provides a useful and intiutive framework for constructing and curating metagenome assebled genomes (MAGs)  

In this tutorial we will use anvi'o to constrcut metagenome assembled genomes (MAGs) from the Saanich Inlet water column with manual refinement. Please visit [anvi'o webpage](http://merenlab.org/software/anvio/) to learn more about the platform and its usage.

After constructing MAGs from several depth intervals you will perform quality control by estimating completion and contamination using [CheckM](https://github.com/Ecogenomics/CheckM) developed by [Parks et al. 2015](https://genome.cshlp.org/content/25/7/1043).

You will then classify medium to high-quality MAGs [Bowers et al. 2017](https://rdcu.be/b9JqO) using [GTDB-Tk](https://github.com/Ecogenomics/GTDBTk) developed by [Chaumeil et al. 2020](https://academic.oup.com/bioinformatics/article/36/6/1925/5626182).

## anvi'o installation

The following instructions based on documentation provided by the Meren lab provide an overview of commands that should work for most operating systems. For detailed instructions please see [anvi'o installation resources](http://merenlab.org/2016/06/26/installation-v2/).

Step 1: Please have a working install of [miniconda](https://docs.conda.io/en/latest/miniconda.html) on your terminal. Conda installation of anvi'o is hassle-free and quick. Make sure the installation is updated by running

```
conda update conda
```

Step 2:

For Mac OSX users

```
# first get a copy of the following file (if you get a "command not found"
# error for wget, you can either install it first, or manually download the
# file from that URL:
wget http://merenlab.org/files/anvio-conda-environments/anvio-environment-6.2.yml

# just to make sure there is not a v6.2 environment already: 
conda env remove --name anvio-6.2

# create a new v6.2 environment using the file you just downloaded:
conda env create -f anvio-environment-6.2.yml

# activate that environment
conda activate anvio-6.2
```

Linux / Windows users:

```
conda create -y --name anvio-6.2 python=3.6
conda activate anvio-6.2
conda install -y -c conda-forge -c bioconda anvio==6.2
```

Step 3: Confirm installation

```
anvi-self-test --suite mini
```

If all is well, your browser should open and load the test run results \n

## Binning tools installation

While we are installing anvi'o, we might as well install binning tools needed to construct MAGs.

Here we install -

1.  METABAT2 developed by [Kang et al. 2019](https://peerj.com/articles/7359/)

2.  MAXBIN2 developed by [Wu et al. 2016](https://academic.oup.com/bioinformatics/article/32/4/605/1744462)

3.  CONCOCT developed by [Alneberg et al. 2014](https://www.nature.com/articles/nmeth.3103)

4.  DASTOOL developed by [Sieber et al. 2018](https://www.nature.com/articles/s41564-018-0171-1)

```
# Metabat2
conda install metabat2

# Maxbin2
conda install maxbin2

# Concoct
conda install -c bioconda concoct

# DAS Tool
conda install -c bioconda das\_tool
```

To verify if these binning tools are accessible by anvi'o, please run these commands:

```
conda activate anvio-6.2
anvi-cluster-contigs --help
````

You should get a help menu on how to use the binning tools in anvi'o. If there is a binning tool that is not accessible, you should see **[NOT FOUND]** next to the binning tool name. If you do not see this tex next to the three binning tools you installed, you are good to proceed.

CheckM (https://anaconda.org/bioconda/checkm-genome) and GTDB-Tk (https://anaconda.org/bioconda/gtdbtk) should also be installed on your terminal.
