# anvi'o binning exercise 

You will be provided with fastq files, contig and merged profile databases to construct MAGs and conduct manual refinement. 

Please see this tutorial on [assembly-based metagenomics](http://merenlab.org/tutorials/assembly-based-metagenomics/) and [metagenomic workflow tutorial](http://merenlab.org/2016/06/22/anvio-tutorial-v2/) on anvi’o to understand how these files were obtained and an in-depth explanation on moving forward. The anvi’o documentation also provides an excellent commentary on the nuances of each step. Additional course material relevant to this tutorial can be found in the lecture slides describing metagenome assembled genomes.  

## Automatic binning on anvi'o using METABAT2, MAXBIN2, and CONCOCT

The following commands provide a step by step approach to obtain MAGs and to load the results on the visualization platform of anvi’o on your browser for manual refinement.

```
#Create variables for profile and contig databases for easy access to files
profile_db = ~/path_to_profile_db 
contig_db = ~/path_to_contig_db 

#Make output directory
Mkdir ~/path_to_output_directory

#Create variable for output directory
bin_out=~/path_to_output_directory

#Operational parameters
threads=32 #Change to suitable amount of threads

#Clustering with metabat2
anvi-cluster-contigs -p $profile_db -c $contig_db -C metabat --driver metabat2 -T $threads --just-do-it

#Clustering with concoct
anvi-cluster-contigs -p $profile_db -c $contig_db -C concoct --driver concoct -T $threads --just-do-it

#Clustering with maxbin2
anvi-cluster-contigs -p $profile_db -c $contig_db -C maxbin --driver maxbin2 -T $threads --just-do-it

#Consensus binning with DASTOOL
anvi-cluster-contigs -p $profile_db -c $contig_db --driver dastool -S metabat,concoct,maxbin -C dastool --just-do-it -T $threads

#Obtain bins as fasta files
anvi-summarize -p $profile_db -c $contig_db -C dastool -o $bin_out/dastool_out

#Copy all bin fasta files for downstream processing such as quality estimation, taxonomic classification, and functional annotation

#Make directory for fasta files for easy access
mkdir $bin_out/fasta_files
cp $bin_out/dastool_out/*/*fasta $bin_out/fasta_files
```

### Note to EDUCE-TA's

In case we want the students to practice manual refinement alone, we can provide them with contaminated MAGs using Concoct.  

The following commands creates a fixed number of MAGs (ideally lower than the actual genomes expected) using Concoct

```
profile_db = ~/path_to_profile_db 
contig_db = ~/path_to_contig_db 

#Make output directory
Mkdir ~/path_to_output_directory

#Operational parameters
threads=32 #Change to suitable amount of threads

bins=N #Change N to half of the number of bins expected to deliberately contaminate bins and later manually refine them

#Clustering with concoct
anvi-cluster-contigs -p $profile_db -c $contig_db -C concoct --driver concoct -T $threads --just-do-it --clusters $bins
```

## Assessing quality MAGs

Check completion and redundancy of MAGs using CheckM. Change paths and operational parameters as required for your system. For a more detailed understanding of how to use CheckM please visit [CheckM wiki](https://github.com/Ecogenomics/CheckM/wiki)

```
#Create checkM output directory
mkdir ~/path_to_checkm_output_dir

#Create variable for output directory
checkm_out=~/path_to_checkm_output_dir

#Establish bins’ fasta files directory 
in_fasta=~/path_to_output_fasta_files

#Operational parameters – change as required on your system
threads=48
pplacer_threads=24

#Run CheckM
checkm lineage_wf -t $threads --pplacer_threads $pplacer_threads \
	-f $checkm_out/checkm_out.tsv --tab_table \
	-x fasta $in_fasta $checkm_out/checkm_lineage_wf
	
#checkm_out.tsv is a tab separated file that provides quality estimates on the set of bins
```

### Note the MAG quality
Open the tab separated file, checkm_out.tsv in a spreadsheet viewer and study the completion and contamination column. We are looking for medium to high-quality MAGs [Bowers et al. 2017](https://rdcu.be/b9JqO).

For this exercise, we aim to have bins with less than 10 % contamination and more than 50 % completion. We will now use the “anvi-refine” feature to manually refine the bins which have more than 10 % contamination. Please make a note of all the bins which are more than 50 % complete and have more than 10 % contamination.

## Manual refinement

Manual refinement on anvi’o is described in great detail [here](http://merenlab.org/2017/05/11/anvi-refine-by-veronika/)

``` {r, echo=FALSE,  results='hide', message=FALSE, warning=FALSE}
knitr::include_app('http://merenlab.org/2017/05/11/anvi-refine-by-veronika/', height = '600px')
```

## Iterative assessment and refinement

After manual refinement of the bins using anvi’o, iteratively run quality assessment and manual refinement of high contamination bins until you have a list of bins which are more than 90 % complete and less than 10 % contaminated. 
