# Processing MAGs 

Taxonomic classification can be done with GTDB-Tk. GTDB-Tk is a toolkit for using the Genome Taxonomy Database (GTDB) to classify taxonomy of the MAGs. Please read [Parks et al. 2018](https://www.nature.com/articles/nbt.4229) to learn more about GTDB and [Chaumeil et al. 2020](https://academic.oup.com/bioinformatics/article/36/6/1925/5626182) for details on the GTDB-Tk tool.

For an in-depth commentary on how to utilize GTDB-Tk please visit https://ecogenomics.github.io/GTDBTk/

After taxonomic classification, functional annotation can be done using TreeSAPP

 You will be implementing a pipeline called Tree-based Sensitive and Accurate Protein Profiler (**TreeSAPP**) for automated reconstruction of the nitrogen cycle along defined redox gradients in Saanich Inlet using the Google Cloud Platform. TreeSAPP takes either metagenomic or metatranscriptomic reads and aligns them to previously binned sequence data with each bin representing a putative microbial taxon.
 
TreeSAPP determines three things:

A. What taxa are in our metagenomic and metatranscriptomic data represented?

B. Which marker genes do these taxa contain (metagenomic data) or actually express (metatranscriptomic data)?

C. At what levels are those genes represented (metagenomic data) or expressed by the taxon?

Please see the [TreeSAPP wiki](https://github.com/hallamlab/TreeSAPP/wiki/Classifying-sequences-with-treesapp-assign) for more information on `treesapp assign`, the subcommand you will use. You will be provided with a script template for both the shell and R portion of your analysis ("treesapp_analysis.sh" and "treesapp_analysis.R") that will guide you as you develop your code.

## Taxonomic classification using GTDB-Tk

The following commands will generate  taxonomic classification of archaeal and bacterial MAGs

```
#Create GTDB-Tk output directory
mkdir ~/path_to_gtdbtk_output_dir
#Create variable for output directory
gtdbtk_out=~/path_to_gtdb_output_dir
#Establish bins’ fasta files directory 
in_fasta=~/path_to_output_fasta_files
#Export GTDB-Tk data path
export GTDBTK_DATA_PATH=~/path_to_GTDB_data
#Operational parameters – change as required on your system
threads=48
pplacer_threads=24
#Run GTDB-Tk
gtdbtk classify_wf -x fa --genome_dir $in_fasta \
	--out_dir $gtdbtk_out \
	--cpus $threads \
	--pplacer_cpus $pplacer_threads
```

You would obtain two soft links in the gtdbtk output directory, each for bacteria and archaea taxonomic classification. Please follow the soft link to locate the actual files for taxonomic classification, and download to your system.


