<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>3 Background information | anvi’o binning and refinement tutorial</title>
  <meta name="description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="3 Background information | anvi’o binning and refinement tutorial" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="3 Background information | anvi’o binning and refinement tutorial" />
  
  <meta name="twitter:description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  

<meta name="author" content="Pranav Sampara, Steven Hallam, Stefanie Christen Sternagel, Julia Anstett, Stephan Koenig" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="instructions.html"/>
<link rel="next" href="introduction-to-anvio.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>



<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">anvi'o binning and refinement tutorial</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#guide-to-the-book"><i class="fa fa-check"></i><b>1.1</b> Guide to the book</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#some-key-features-of-metagenomic-analysis"><i class="fa fa-check"></i><b>1.2</b> Some key features of metagenomic analysis</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#basic-outline-of-a-metagenomics-pipeline"><i class="fa fa-check"></i><b>1.3</b> Basic outline of a metagenomics pipeline</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="instructions.html"><a href="instructions.html"><i class="fa fa-check"></i><b>2</b> Instructions</a><ul>
<li class="chapter" data-level="2.1" data-path="instructions.html"><a href="instructions.html#goal-of-the-exercise"><i class="fa fa-check"></i><b>2.1</b> Goal of the exercise</a></li>
<li class="chapter" data-level="2.2" data-path="instructions.html"><a href="instructions.html#project-ideas"><i class="fa fa-check"></i><b>2.2</b> Project ideas</a></li>
<li class="chapter" data-level="2.3" data-path="instructions.html"><a href="instructions.html#reports-and-assessment"><i class="fa fa-check"></i><b>2.3</b> Reports and Assessment</a><ul>
<li class="chapter" data-level="2.3.1" data-path="instructions.html"><a href="instructions.html#abstract-250-word-max."><i class="fa fa-check"></i><b>2.3.1</b> Abstract (250 word max.)</a></li>
<li class="chapter" data-level="2.3.2" data-path="instructions.html"><a href="instructions.html#importance-120-word-max."><i class="fa fa-check"></i><b>2.3.2</b> Importance (120 word max.)</a></li>
<li class="chapter" data-level="2.3.3" data-path="instructions.html"><a href="instructions.html#introduction-1250-words"><i class="fa fa-check"></i><b>2.3.3</b> Introduction (1250 words)</a></li>
<li class="chapter" data-level="2.3.4" data-path="instructions.html"><a href="instructions.html#methods-750-words"><i class="fa fa-check"></i><b>2.3.4</b> Methods (750 words)</a></li>
<li class="chapter" data-level="2.3.5" data-path="instructions.html"><a href="instructions.html#results-1000-words"><i class="fa fa-check"></i><b>2.3.5</b> Results (1000 words)</a></li>
<li class="chapter" data-level="2.3.6" data-path="instructions.html"><a href="instructions.html#discussion-1000-words"><i class="fa fa-check"></i><b>2.3.6</b> Discussion (1000 words)</a></li>
<li class="chapter" data-level="2.3.7" data-path="instructions.html"><a href="instructions.html#figures-and-tables-4-with-captions"><i class="fa fa-check"></i><b>2.3.7</b> Figures and tables (≥4 with captions)</a></li>
<li class="chapter" data-level="2.3.8" data-path="instructions.html"><a href="instructions.html#references-20"><i class="fa fa-check"></i><b>2.3.8</b> References (&gt;20)</a></li>
<li class="chapter" data-level="2.3.9" data-path="instructions.html"><a href="instructions.html#fin."><i class="fa fa-check"></i><b>2.3.9</b> Fin.</a></li>
<li class="chapter" data-level="2.3.10" data-path="instructions.html"><a href="instructions.html#ujemi-submission"><i class="fa fa-check"></i><b>2.3.10</b> UJEMI submission</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="background-information.html"><a href="background-information.html"><i class="fa fa-check"></i><b>3</b> Background information</a><ul>
<li class="chapter" data-level="3.1" data-path="background-information.html"><a href="background-information.html#project-information"><i class="fa fa-check"></i><b>3.1</b> Project information</a></li>
<li class="chapter" data-level="3.2" data-path="background-information.html"><a href="background-information.html#data-collection"><i class="fa fa-check"></i><b>3.2</b> Data collection</a></li>
<li class="chapter" data-level="3.3" data-path="background-information.html"><a href="background-information.html#data-availability"><i class="fa fa-check"></i><b>3.3</b> Data availability</a></li>
<li class="chapter" data-level="3.4" data-path="background-information.html"><a href="background-information.html#preliminary-data-processing"><i class="fa fa-check"></i><b>3.4</b> Preliminary data processing</a><ul>
<li class="chapter" data-level="3.4.1" data-path="background-information.html"><a href="background-information.html#quality-filtering-of-reads"><i class="fa fa-check"></i><b>3.4.1</b> Quality filtering of reads</a></li>
<li class="chapter" data-level="3.4.2" data-path="background-information.html"><a href="background-information.html#assembly"><i class="fa fa-check"></i><b>3.4.2</b> Assembly</a></li>
<li class="chapter" data-level="3.4.3" data-path="background-information.html"><a href="background-information.html#deduplication"><i class="fa fa-check"></i><b>3.4.3</b> Deduplication</a></li>
<li class="chapter" data-level="3.4.4" data-path="background-information.html"><a href="background-information.html#mapping"><i class="fa fa-check"></i><b>3.4.4</b> Mapping</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="introduction-to-anvio.html"><a href="introduction-to-anvio.html"><i class="fa fa-check"></i><b>4</b> Introduction to anvi’o</a><ul>
<li class="chapter" data-level="4.1" data-path="introduction-to-anvio.html"><a href="introduction-to-anvio.html#what-is-anvio"><i class="fa fa-check"></i><b>4.1</b> What is anvi’o</a></li>
<li class="chapter" data-level="4.2" data-path="introduction-to-anvio.html"><a href="introduction-to-anvio.html#anvio-installation"><i class="fa fa-check"></i><b>4.2</b> anvi’o installation</a></li>
<li class="chapter" data-level="4.3" data-path="introduction-to-anvio.html"><a href="introduction-to-anvio.html#binning-tools-installation"><i class="fa fa-check"></i><b>4.3</b> Binning tools installation</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="anvio-binning-exercise.html"><a href="anvio-binning-exercise.html"><i class="fa fa-check"></i><b>5</b> anvi’o binning exercise</a><ul>
<li class="chapter" data-level="5.1" data-path="anvio-binning-exercise.html"><a href="anvio-binning-exercise.html#automatic-binning-on-anvio-using-metabat2-maxbin2-and-concoct"><i class="fa fa-check"></i><b>5.1</b> Automatic binning on anvi’o using METABAT2, MAXBIN2, and CONCOCT</a><ul>
<li class="chapter" data-level="5.1.1" data-path="anvio-binning-exercise.html"><a href="anvio-binning-exercise.html#note-to-educe-tas"><i class="fa fa-check"></i><b>5.1.1</b> Note to EDUCE-TA’s</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="anvio-binning-exercise.html"><a href="anvio-binning-exercise.html#assessing-quality-mags"><i class="fa fa-check"></i><b>5.2</b> Assessing quality MAGs</a><ul>
<li class="chapter" data-level="5.2.1" data-path="anvio-binning-exercise.html"><a href="anvio-binning-exercise.html#note-the-mag-quality"><i class="fa fa-check"></i><b>5.2.1</b> Note the MAG quality</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="anvio-binning-exercise.html"><a href="anvio-binning-exercise.html#manual-refinement"><i class="fa fa-check"></i><b>5.3</b> Manual refinement</a></li>
<li class="chapter" data-level="5.4" data-path="anvio-binning-exercise.html"><a href="anvio-binning-exercise.html#iterative-assessment-and-refinement"><i class="fa fa-check"></i><b>5.4</b> Iterative assessment and refinement</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="processing-mags.html"><a href="processing-mags.html"><i class="fa fa-check"></i><b>6</b> Processing MAGs</a><ul>
<li class="chapter" data-level="6.1" data-path="processing-mags.html"><a href="processing-mags.html#taxonomic-classification-using-gtdb-tk"><i class="fa fa-check"></i><b>6.1</b> Taxonomic classification using GTDB-Tk</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="protocol-and-scripts.html"><a href="protocol-and-scripts.html"><i class="fa fa-check"></i><b>7</b> Protocol and scripts</a><ul>
<li class="chapter" data-level="7.1" data-path="protocol-and-scripts.html"><a href="protocol-and-scripts.html#project-files"><i class="fa fa-check"></i><b>7.1</b> Project files</a></li>
<li class="chapter" data-level="7.2" data-path="protocol-and-scripts.html"><a href="protocol-and-scripts.html#quality-filtering-and-assembly"><i class="fa fa-check"></i><b>7.2</b> Quality filtering and assembly</a></li>
<li class="chapter" data-level="7.3" data-path="protocol-and-scripts.html"><a href="protocol-and-scripts.html#deduplication-of-contigs"><i class="fa fa-check"></i><b>7.3</b> Deduplication of contigs</a></li>
<li class="chapter" data-level="7.4" data-path="protocol-and-scripts.html"><a href="protocol-and-scripts.html#prokka-annotation"><i class="fa fa-check"></i><b>7.4</b> Prokka annotation</a></li>
<li class="chapter" data-level="7.5" data-path="protocol-and-scripts.html"><a href="protocol-and-scripts.html#mapping-1"><i class="fa fa-check"></i><b>7.5</b> Mapping</a></li>
<li class="chapter" data-level="7.6" data-path="protocol-and-scripts.html"><a href="protocol-and-scripts.html#create-anvio-databases"><i class="fa fa-check"></i><b>7.6</b> Create anvi’o databases</a><ul>
<li class="chapter" data-level="7.6.1" data-path="protocol-and-scripts.html"><a href="protocol-and-scripts.html#building-contigs-database"><i class="fa fa-check"></i><b>7.6.1</b> Building contigs database</a></li>
<li class="chapter" data-level="7.6.2" data-path="protocol-and-scripts.html"><a href="protocol-and-scripts.html#building-profile-databases"><i class="fa fa-check"></i><b>7.6.2</b> Building profile databases</a></li>
</ul></li>
<li class="chapter" data-level="7.7" data-path="protocol-and-scripts.html"><a href="protocol-and-scripts.html#binning"><i class="fa fa-check"></i><b>7.7</b> Binning</a></li>
<li class="chapter" data-level="7.8" data-path="protocol-and-scripts.html"><a href="protocol-and-scripts.html#assess-genome-quality-with-checkm-and-manual-refinement"><i class="fa fa-check"></i><b>7.8</b> Assess genome quality with CheckM and manual refinement</a><ul>
<li class="chapter" data-level="7.8.1" data-path="protocol-and-scripts.html"><a href="protocol-and-scripts.html#checkm-usage"><i class="fa fa-check"></i><b>7.8.1</b> CheckM usage</a></li>
<li class="chapter" data-level="7.8.2" data-path="protocol-and-scripts.html"><a href="protocol-and-scripts.html#manual-refinement-1"><i class="fa fa-check"></i><b>7.8.2</b> Manual refinement</a></li>
<li class="chapter" data-level="7.8.3" data-path="protocol-and-scripts.html"><a href="protocol-and-scripts.html#iterative-assessment-and-refinement-1"><i class="fa fa-check"></i><b>7.8.3</b> Iterative assessment and refinement</a></li>
<li class="chapter" data-level="7.8.4" data-path="protocol-and-scripts.html"><a href="protocol-and-scripts.html#store-bins-on-the-server"><i class="fa fa-check"></i><b>7.8.4</b> Store bins on the server</a></li>
</ul></li>
<li class="chapter" data-level="7.9" data-path="protocol-and-scripts.html"><a href="protocol-and-scripts.html#assigning-gtdb-taxonomy-to-your-bins"><i class="fa fa-check"></i><b>7.9</b> Assigning GTDB taxonomy to your bins</a></li>
<li class="chapter" data-level="7.10" data-path="protocol-and-scripts.html"><a href="protocol-and-scripts.html#make-sure-to-echo-the-gtdb-tk-data-pathecho"><i class="fa fa-check"></i><b>7.10</b> Make sure to echo the GTDB-Tk data pathecho</a></li>
<li class="chapter" data-level="7.11" data-path="protocol-and-scripts.html"><a href="protocol-and-scripts.html#working-with-treesapp"><i class="fa fa-check"></i><b>7.11</b> Working with TreeSAPP</a></li>
<li class="chapter" data-level="7.12" data-path="protocol-and-scripts.html"><a href="protocol-and-scripts.html#treesapp-script"><i class="fa fa-check"></i><b>7.12</b> TreeSAPP script</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i><b>8</b> References</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">anvi’o binning and refinement tutorial</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="background-information" class="section level1">
<h1><span class="header-section-number">3</span> Background information</h1>
<div id="project-information" class="section level2">
<h2><span class="header-section-number">3.1</span> Project information</h2>
<p>Marine oxygen minimum zones (OMZ) are widespread areas of low dissolved oxygen (DO) in subsurface waters. Climate change resulting in increased stratification and reduced oxygen solubility in warming waters leads to an expansion of OMZ. Consequently as oxygen levels decline, the microbial communities inhabiting OMZ waters shift their metabolisms to utilize alternative terminal electron acceptors. This results in the production of climate active trace gases such as nitrous oxide (N2O) and methane (CH4). OMZs provide useful environmental contexts in which to study coupled biogeochemical cycling through microbial metabolic networks. By combining ’omics data with biogeochemical data from OMZs it becomes possible to evaluate regulatory and response dynamics of microbial communities to changing DO levels. In this study, Saanich inlet, a seasonally anoxic fjord on the coast of Vancouver Island British Columbia, was used as a model ecosystem to study microbial community responses to changing DO levels.</p>
</div>
<div id="data-collection" class="section level2">
<h2><span class="header-section-number">3.2</span> Data collection</h2>
<p>Multi-omic sequence information (DNA, RNA and proteins) from Saanich Inlet along with geochemical parameter information was collected over many years prodicing a time series record of recurring water column stratificaiton and deep water renewal. For this tutorial, 90 metagenomes spanning water colum redix gradients in space and time, totalling 4.1 TB of cleaned reads or 16.2 GB of assembled data, were prepared. <a href="data_multi-omic_samples.png">(a) Oxygen concentration contour for CTD data (February 2008 onward)35 indicating 16 sampling depths for water column geochemistry and high-resolution (HR) DNA samples for SSU libraries (small black dots) and six major depths for large volume (LV) samples for meta-genomics, -transcriptomics, -proteomics and LV SSU libraries (large black dots). (b) Sample inventory from February 2006 to October 2014 indicating multi-omic datasets included in this manuscript (solid black), in previous publications (gray) and accompanying datasets currently undergoing processing and analysis (open gray).</a></p>
<p>Please refer <a href="https://www.nature.com/articles/sdata2017160">Hawley, A. K. et al. (2017)</a> and <a href="https://www.nature.com/articles/sdata2017159">Torres-Beltrán, M. et al. (2017)</a></p>
</div>
<div id="data-availability" class="section level2">
<h2><span class="header-section-number">3.3</span> Data availability</h2>
<p>The following files are available on Orca.</p>
<pre><code># Directory with all files
/projects/micb405/resources/metagenomics_data

# Raw reads by depth
/projects/micb405/resources/metagenomics_data/SI072_100m_raw_reads.fastq.gz
/projects/micb405/resources/metagenomics_data/SI072_120m_raw_reads.fastq.gz
/projects/micb405/resources/metagenomics_data/SI072_200m_raw_reads.fastq.gz

# Deduplicated set of contigs from the combined 100, 120 and 200 m datasets
/projects/micb405/resources/metagenomics_data/SI072_Dedupe_Contigs.fa

# Directory with bins refined with anvi&#39;o and CheckM table
/projects/micb405/resources/metagenomics_data/SI072_Bins_Refined_final

# Directory with taxonomic classification of refined bins by GTDB-Tk
/projects/micb405/resources/metagenomics_data/SI072_Bins_Ref_5_GTDB_r95

# Directory with taxonomic classification of refined bins by Centrifuge
/projects/micb405/resources/metagenomics_data/Import_To_Contig_DB_files/Centrifuge_Taxonomy

# Directory with Prokka output for MAGs
/projects/micb405/resources/metagenomics_data/Import_To_Contig_DB_files/Prokka_Functions

# BAM files and indices for data
/projects/micb405/resources/metagenomics_data/bams</code></pre>
</div>
<div id="preliminary-data-processing" class="section level2">
<h2><span class="header-section-number">3.4</span> Preliminary data processing</h2>
<div id="quality-filtering-of-reads" class="section level3">
<h3><span class="header-section-number">3.4.1</span> Quality filtering of reads</h3>
<p>Quality filtering was performed with Trimmomtic (v.0.35) developed by <a href="https://doi.org/10.1093/bioinformatics/btu170">Bolger et al., (2014)</a>. Please check <a href="http://www.usadellab.org/cms/?page=trimmomatic">Trimmomatic documentation</a> for more details on the use of Trimmomatic. The following parameters were used for Trimmomatic, with the other parameters as defaults:
LEADING:3
TRAILING:3
SLIDINGWINDOW:4:15
MINLEN:36</p>
</div>
<div id="assembly" class="section level3">
<h3><span class="header-section-number">3.4.2</span> Assembly</h3>
<p>Megahit (v.1.1.3), developed by <a href="https://doi.org/10.1093/bioinformatics/btv033">Li et al., (2015)</a>, was used with the following parameters (the rest of the parameters being default):
–k-min 27
–k-step 10
–min-contig-len 500</p>
<p>Please review <a href="https://github.com/voutcn/megahit/wiki/Assembly-Tips">megahit documentation</a> for more details on the usage</p>
</div>
<div id="deduplication" class="section level3">
<h3><span class="header-section-number">3.4.3</span> Deduplication</h3>
<p><a href="https://jgi.doe.gov/data-and-tools/bbtools/">BBtools</a> and dedupe.sh script were used to deduplicate contigs
Please visit the <a href="https://jgi.doe.gov/data-and-tools/bbtools/bb-tools-user-guide/dedupe-guide/#:~:text=Dedupe%20was%20written%20to%20eliminate,of%20substitutions%20or%20edit%20distance.&amp;text=Dedupe%20is%20primarily%20designed%20to%20handle%20these%20situations.">deduplication guide</a> from the Joint Genome Institute for more information.</p>
<p>The following command generates deduplicated assemblies</p>
<pre><code>dedupe.sh in=SI072_Combined_contigs.fa out=SI072_Dedupe_Contigs.fa outd=duplicates.fa</code></pre>
</div>
<div id="mapping" class="section level3">
<h3><span class="header-section-number">3.4.4</span> Mapping</h3>
<p>Mapping was performed with bwa-mem (<a href="https://arxiv.org/pdf/1303.3997.pdf">Li H. (2013)</a>). Please check <a href="https://github.com/lh3/bwa">bwa-mem documentation</a> for details on the tool’s usage</p>
<pre><code>bwa mem -t 4 SI072_Dedupe_Contigs.fa SI072_(depth_here)m_raw_reads.fastq.gz | samtools view -b | samtools sort -o SI072_(depth_here)m_Dedupe_contigs_sorted.bam

samtools view -b -F 4 SI072_(depth_here)m_Dedupe_contigs_sorted.bam &gt; SI072_(depth_here)m_Dedupe_contigs_sorted.mapped.bam
</code></pre>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="instructions.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="introduction-to-anvio.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "section"
},
"link": "GitHub repo link"
});
});
</script>

</body>

</html>
