<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>2 Instructions | anvi’o binning and refinement tutorial</title>
  <meta name="description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="2 Instructions | anvi’o binning and refinement tutorial" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="2 Instructions | anvi’o binning and refinement tutorial" />
  
  <meta name="twitter:description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  

<meta name="author" content="Pranav Sampara and Steven Hallam" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="index.html"/>
<link rel="next" href="background-information.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<link href="libs/anchor-sections-1.0/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0/anchor-sections.js"></script>



<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Anvio binning and refinement tutorial</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#guide-to-the-book"><i class="fa fa-check"></i><b>1.1</b> Guide to the book</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#brief-introduction-to-omics-methods"><i class="fa fa-check"></i><b>1.2</b> Brief introduction to ’omics methods</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#some-key-features-of-metagenomic-analysis"><i class="fa fa-check"></i><b>1.3</b> Some key features of metagenomic analysis</a></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#key-points-to-be-aware-of-and-known-pitfalls-in-metagenomics"><i class="fa fa-check"></i><b>1.4</b> Key points to be aware of and known pitfalls in metagenomics</a></li>
<li class="chapter" data-level="1.5" data-path="index.html"><a href="index.html#basic-outline-of-a-metagenomics-pipeline"><i class="fa fa-check"></i><b>1.5</b> Basic outline of a metagenomics pipeline</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="instructions.html"><a href="instructions.html"><i class="fa fa-check"></i><b>2</b> Instructions</a><ul>
<li class="chapter" data-level="2.1" data-path="instructions.html"><a href="instructions.html#goal-of-the-exercise"><i class="fa fa-check"></i><b>2.1</b> Goal of the exercise</a></li>
<li class="chapter" data-level="2.2" data-path="instructions.html"><a href="instructions.html#project-ideas"><i class="fa fa-check"></i><b>2.2</b> Project ideas</a></li>
<li class="chapter" data-level="2.3" data-path="instructions.html"><a href="instructions.html#reports-and-assessment"><i class="fa fa-check"></i><b>2.3</b> Reports and Assessment</a><ul>
<li class="chapter" data-level="2.3.1" data-path="instructions.html"><a href="instructions.html#abstract-250-word-max."><i class="fa fa-check"></i><b>2.3.1</b> Abstract (250 word max.)</a></li>
<li class="chapter" data-level="2.3.2" data-path="instructions.html"><a href="instructions.html#importance-120-word-max."><i class="fa fa-check"></i><b>2.3.2</b> Importance (120 word max.)</a></li>
<li class="chapter" data-level="2.3.3" data-path="instructions.html"><a href="instructions.html#introduction-1250-words"><i class="fa fa-check"></i><b>2.3.3</b> Introduction (1250 words)</a></li>
<li class="chapter" data-level="2.3.4" data-path="instructions.html"><a href="instructions.html#methods-750-words"><i class="fa fa-check"></i><b>2.3.4</b> Methods (750 words)</a></li>
<li class="chapter" data-level="2.3.5" data-path="instructions.html"><a href="instructions.html#results-1000-words"><i class="fa fa-check"></i><b>2.3.5</b> Results (1000 words)</a></li>
<li class="chapter" data-level="2.3.6" data-path="instructions.html"><a href="instructions.html#discussion-1000-words"><i class="fa fa-check"></i><b>2.3.6</b> Discussion (1000 words)</a></li>
<li class="chapter" data-level="2.3.7" data-path="instructions.html"><a href="instructions.html#figures-and-tables-4-with-captions"><i class="fa fa-check"></i><b>2.3.7</b> Figures and tables (≥4 with captions)</a></li>
<li class="chapter" data-level="2.3.8" data-path="instructions.html"><a href="instructions.html#references-20"><i class="fa fa-check"></i><b>2.3.8</b> References (&gt;20)</a></li>
<li class="chapter" data-level="2.3.9" data-path="instructions.html"><a href="instructions.html#fin."><i class="fa fa-check"></i><b>2.3.9</b> Fin.</a></li>
<li class="chapter" data-level="2.3.10" data-path="instructions.html"><a href="instructions.html#ujemi-submission"><i class="fa fa-check"></i><b>2.3.10</b> UJEMI submission</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="background-information.html"><a href="background-information.html"><i class="fa fa-check"></i><b>3</b> Background information</a><ul>
<li class="chapter" data-level="3.1" data-path="background-information.html"><a href="background-information.html#project-information"><i class="fa fa-check"></i><b>3.1</b> Project information</a></li>
<li class="chapter" data-level="3.2" data-path="background-information.html"><a href="background-information.html#data-collection"><i class="fa fa-check"></i><b>3.2</b> Data collection</a></li>
<li class="chapter" data-level="3.3" data-path="background-information.html"><a href="background-information.html#data-availability"><i class="fa fa-check"></i><b>3.3</b> Data availability</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="introduction-to-anvio.html"><a href="introduction-to-anvio.html"><i class="fa fa-check"></i><b>4</b> Introduction to anvi’o</a><ul>
<li class="chapter" data-level="4.1" data-path="introduction-to-anvio.html"><a href="introduction-to-anvio.html#what-is-anvio"><i class="fa fa-check"></i><b>4.1</b> What is anvi’o</a></li>
<li class="chapter" data-level="4.2" data-path="introduction-to-anvio.html"><a href="introduction-to-anvio.html#anvio-installation"><i class="fa fa-check"></i><b>4.2</b> anvi’o installation</a></li>
<li class="chapter" data-level="4.3" data-path="introduction-to-anvio.html"><a href="introduction-to-anvio.html#binning-tools-installation"><i class="fa fa-check"></i><b>4.3</b> Binning tools installation</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="anvio-binning-exercise.html"><a href="anvio-binning-exercise.html"><i class="fa fa-check"></i><b>5</b> anvi’o binning exercise</a><ul>
<li class="chapter" data-level="5.1" data-path="anvio-binning-exercise.html"><a href="anvio-binning-exercise.html#automatic-binning-on-anvio-using-metabat2-maxbin2-and-concoct"><i class="fa fa-check"></i><b>5.1</b> Automatic binning on anvi’o using METABAT2, MAXBIN2, and CONCOCT</a><ul>
<li class="chapter" data-level="5.1.1" data-path="anvio-binning-exercise.html"><a href="anvio-binning-exercise.html#note-to-educe-tas"><i class="fa fa-check"></i><b>5.1.1</b> Note to EDUCE-TA’s</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="anvio-binning-exercise.html"><a href="anvio-binning-exercise.html#assessing-quality-mags"><i class="fa fa-check"></i><b>5.2</b> Assessing quality MAGs</a><ul>
<li class="chapter" data-level="5.2.1" data-path="anvio-binning-exercise.html"><a href="anvio-binning-exercise.html#note-the-mag-quality"><i class="fa fa-check"></i><b>5.2.1</b> Note the MAG quality</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="anvio-binning-exercise.html"><a href="anvio-binning-exercise.html#manual-refinement"><i class="fa fa-check"></i><b>5.3</b> Manual refinement</a></li>
<li class="chapter" data-level="5.4" data-path="anvio-binning-exercise.html"><a href="anvio-binning-exercise.html#iterative-assessment-and-refinement"><i class="fa fa-check"></i><b>5.4</b> Iterative assessment and refinement</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="processing-mags.html"><a href="processing-mags.html"><i class="fa fa-check"></i><b>6</b> Processing MAGs</a><ul>
<li class="chapter" data-level="6.1" data-path="processing-mags.html"><a href="processing-mags.html#taxonomic-classification-using-gtdb-tk"><i class="fa fa-check"></i><b>6.1</b> Taxonomic classification using GTDB-Tk</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i><b>7</b> References</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">anvi’o binning and refinement tutorial</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="instructions" class="section level1">
<h1><span class="header-section-number">2</span> Instructions</h1>
<div id="goal-of-the-exercise" class="section level2">
<h2><span class="header-section-number">2.1</span> Goal of the exercise</h2>
<p>The primary objective of this tutorial is to use anvi’o to construct population genome bins, also known as metagenome assembled genomes (MAGs), to identify the taxonomy and metabolic potential of medium to high quality MAGs of interest <a href="https://rdcu.be/b9JqO">Bowers et al. 2017</a>.</p>
<p>You will be working with metagenomic data sets from the Saanich Inlet time series Cruise #72. Prerequisite input files are provided to construct individual contig and merged profile databases for binning. As indicated above you will be using anvi’o, an ‘omics analysis and visualization platform’ developed by the Meren lab at the University of Chicago. Specific automated binning applications invoked by anvi’o can include MetaBAT2, MaxBin2, CONCOCT, and DAS Tool.</p>
<p>Following MAG construciton you will use CheckM to check MAG completeness and contamination and GTDB-Tk to assign taxonomic affiliation. Collectively this information will enable you to reconstruct metabolic networks at the individual, population and community levels of biological organization. You will also be using prokka to predict and annotate open reading frames, and BWA to map reads onto MAGs to determine coverage and assess how representative the bins in the sampled community.</p>
<p>Results from anvi’o will be compared to those generated using TreeSAPP, an application for fast phylogentic mapping of functional anchor genes, and the subject of a separate tutorial.</p>
</div>
<div id="project-ideas" class="section level2">
<h2><span class="header-section-number">2.2</span> Project ideas</h2>
<p>Note that these are guiding ideas only. You are welcome to follow your own interests in developing a cogent reporting plan. You are working with primary scientific data and are encouraged to let the science “take you somewhere”.</p>
<ul>
<li>Determine what fraction of reads map to MAGs versus contigs not in MAGs. Are medium to high quality MAGs more or less abundant in the sample based on read mapping?</li>
<li>Identify nitrogen cycling genes encoded in MAGs using prokka annotations and compare these results to TreeSAPP annotations using both MAGs and contigs. Are the results equivalent or different?</li>
<li>Describe GTDB-Tk assigned taxonomies of MAGs. Can you identify previously described microbial groups based on the published record?</li>
<li>Evaluate changes in nitrogen cycling gene taxonomy and abundance at different depths in the Saanich inlet water column with special emphasis on the modular denitrification pathway. How does the frequency distribution of denitrification genes change in relation to oxygen concentration and are there any gaps in pathway representation that are more likely to be filled by a specific taxonoic group?</li>
</ul>
</div>
<div id="reports-and-assessment" class="section level2">
<h2><span class="header-section-number">2.3</span> Reports and Assessment</h2>
<p>Reports should be formatted as per the <a href="https://jb.asm.org/sites/default/files/additional-assets/JB-ITA.pdf">Instructions to Authors</a> for the <a href="https://jb.asm.org/">Journal of Bacteriology</a>. These are roughly equivalent to the JEMI+ reporting guidelines (<a href="https://jemi.microbiology.ubc.ca/instructions/submission-guidelines" class="uri">https://jemi.microbiology.ubc.ca/instructions/submission-guidelines</a>).</p>
<p>Reports will account for 20% of the course marks (50 points). The project will be assessed by the following criteria:</p>
<ol style="list-style-type: decimal">
<li>Completion of the proposed workflow. Expected outputs are available and formatted correctly (40%).</li>
<li>A written report (4000 - 4500 words) adhering to the structure below (40%).</li>
<li>Five minute oral presentation following the structure of the written report with emphasis on visual analysis of results (20%).</li>
</ol>
<p>Each group will complete <strong>one</strong> report with the sections listed below. The word counts for each section are recommnedations and can be adjusted as needed, but the topics within each must needs be included!</p>
<div id="abstract-250-word-max." class="section level3">
<h3><span class="header-section-number">2.3.1</span> Abstract (250 word max.)</h3>
<ul>
<li>Concisely summarize the major results and implications of the report.</li>
</ul>
</div>
<div id="importance-120-word-max." class="section level3">
<h3><span class="header-section-number">2.3.2</span> Importance (120 word max.)</h3>
<ul>
<li>Lay explanation of the significance of the research performed to the field of environmental microbiology.</li>
</ul>
</div>
<div id="introduction-1250-words" class="section level3">
<h3><span class="header-section-number">2.3.3</span> Introduction (1250 words)</h3>
<ul>
<li>Introduce Saanich Inlet as a model ecosystem for studying microbial community responses to ocean deoxygenation e.g. seasonal cycles, relevant biogeochemistry, nitrogen cycle, relationship to oxygen minimum zones, previous studies, etc.</li>
<li>Provide relevant information regarding metagenomic approaches (data generation and application), pros, cons, and alternatives.</li>
<li>Introduce one or all of: metagenomic binning, genome assembly, taxonomic profiling.</li>
<li>Justify reasoning for using cultivation-independent methods to study microbial metabolic networks e.g. nitrogen cycle.</li>
<li>Here we describe “X, Y, Z”. End this section by telling the reader what you are going to tell them.</li>
</ul>
</div>
<div id="methods-750-words" class="section level3">
<h3><span class="header-section-number">2.3.4</span> Methods (750 words)</h3>
<ul>
<li>Explain the workflow to generate inputs for the project to show you understand each step.</li>
<li>Explain the tools used at each step, and why they were chosen, to answer your group’s specific question.</li>
<li>Reference the bash and R(md) scripts containing all commands. Do not include the scripts themselves in the text of your manuscript but submit them as individual files on Canvas.</li>
<li>Explain why each step was used, what the outputs are, etc. For example, metagenome assembly was used to build contigs from the quality controlled sequencing reads and used as inputs for binning. <strong>Don’t be silly and plagiarise this</strong>.</li>
<li>Explain any potential deviations from the default parameters or the suggested workflow.</li>
</ul>
</div>
<div id="results-1000-words" class="section level3">
<h3><span class="header-section-number">2.3.5</span> Results (1000 words)</h3>
<ul>
<li>Provide a summary of the binning results, CheckM, and GTDB-Tk outputs, as well as those from other software used, if applicable. Explain rationale for focusing on high-quality MAGs but provide information on the total number of bins constructed and their relative abundance based on read mapping.</li>
<li>Report on the Prokka annotations such as rRNA predicted, # genes annotated, and potential metabolisms encoded in the medium to high-quality MAGs.</li>
<li>Describe the relevant results from your group’s selected project ideas. Explain your experimental logic for each step in the analysis workflow.</li>
</ul>
</div>
<div id="discussion-1000-words" class="section level3">
<h3><span class="header-section-number">2.3.6</span> Discussion (1000 words)</h3>
<ul>
<li>Relate your results back to the introduction. Think about the environmental context and how the obseved taxonmy and function are related to measured geochemical conditions in the Saanich Inlet water column. Consider how each MAG contributes to different metabolic interactions (e.g. denitrification, sulfur oxidation, amino acid biosynthesis, etc). Can you find evidence for distributed metabolism?</li>
<li>How do your results compare to previous studies? Are you able to make new observations that expand on prior knowledge?</li>
<li>Briefly state any challenges encountered (we know there will be some!) and troubleshooting (parameter or software changes). Do new questions arise from your analysis? If you were to continue working on this proejct what future directions might you take to expand on the work or test specific hypotheses?</li>
</ul>
</div>
<div id="figures-and-tables-4-with-captions" class="section level3">
<h3><span class="header-section-number">2.3.7</span> Figures and tables (≥4 with captions)</h3>
<p>Some recommended figures to include:</p>
<ul>
<li>Geochemical gradients for nutrients (phosphate, silicate), nitrogen compounds (nitrate, nitrite, ammonia, nitrous oxide), oxygen and/or sulfur compounds. These are depth-dependent.</li>
<li>Scatter plot comparing contamination % (Y-axis) versus completion % (X-axis) between population genome bins. Consider colour coding the bins by taxonomic rank e.g. phylum, order, etc. Indicate high, medium and low contamination bins (see Canvas for an example).</li>
<li>RPKM bubble-plot of genes in a pathway versus taxonomy.</li>
<li>A diagram showing the taxonomic identity and abundance for each step in the denitrification pathway.</li>
<li>A panel of phylogenetic trees showing the depth specific trends for selected TreeSAPP reference packages.</li>
</ul>
</div>
<div id="references-20" class="section level3">
<h3><span class="header-section-number">2.3.8</span> References (&gt;20)</h3>
<p>Necessary citations include:</p>
<ul>
<li>Each software application used (including those in the provided bash scripts (MEGAHIT, MetaBAT, CheckM, etc.). Make sure you include version numbers in your methods and indicate parameter settings used e.g. default.</li>
<li>Relevant papers from Saanich Inlet or other marine oxygen minimum zones.</li>
</ul>
</div>
<div id="fin." class="section level3">
<h3><span class="header-section-number">2.3.9</span> Fin.</h3>
<p><strong>This evaluation rubric should serve as a useful scaffold for your group. How you expand on it will depend on your combined interests and analytic choices. Let the science take you somewhere!</strong></p>
</div>
<div id="ujemi-submission" class="section level3">
<h3><span class="header-section-number">2.3.10</span> UJEMI submission</h3>
<p>Outstanding reports will be invited to submit to The Undergraduate Journal of Experimental Microbiology and Immunology (<a href="https://jemi.microbiology.ubc.ca/">UJEMI</a>).</p>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="index.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="background-information.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "section"
},
"link": "GitHub repo link"
});
});
</script>

</body>

</html>
