<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>5 Anvio binning exercise | Introduction</title>
  <meta name="description" content="5 Anvio binning exercise | Introduction" />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="5 Anvio binning exercise | Introduction" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="5 Anvio binning exercise | Introduction" />
  
  
  

<meta name="author" content="Pranav_Sampara" />


<meta name="date" content="2020-05-11" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="introduction-to-anvio.html"/>
<link rel="next" href="processing-mags.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<link href="libs/anchor-sections-1.0/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0/anchor-sections.js"></script>



</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>1</b> Introduction</a><ul>
<li class="chapter" data-level="1.1" data-path="introduction.html"><a href="introduction.html#guide-to-the-book"><i class="fa fa-check"></i><b>1.1</b> Guide to the book</a></li>
<li class="chapter" data-level="1.2" data-path="introduction.html"><a href="introduction.html#brief-introduction-to-omics-methods"><i class="fa fa-check"></i><b>1.2</b> Brief introduction to ’omics methods</a></li>
<li class="chapter" data-level="1.3" data-path="introduction.html"><a href="introduction.html#some-key-aspects-of-metagenomics"><i class="fa fa-check"></i><b>1.3</b> Some key aspects of metagenomics</a></li>
<li class="chapter" data-level="1.4" data-path="introduction.html"><a href="introduction.html#key-points-to-be-aware-of-and-known-pitfalls-in-metagenomics"><i class="fa fa-check"></i><b>1.4</b> Key points to be aware of and known pitfalls in metagenomics</a></li>
<li class="chapter" data-level="1.5" data-path="introduction.html"><a href="introduction.html#basic-outline-of-a-metagenomics-pipeline"><i class="fa fa-check"></i><b>1.5</b> Basic outline of a metagenomics pipeline</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="metagenomics-workflow.html"><a href="metagenomics-workflow.html"><i class="fa fa-check"></i><b>2</b> Metagenomics workflow</a></li>
<li class="chapter" data-level="3" data-path="background-information.html"><a href="background-information.html"><i class="fa fa-check"></i><b>3</b> Background information</a><ul>
<li class="chapter" data-level="3.1" data-path="background-information.html"><a href="background-information.html#project-information"><i class="fa fa-check"></i><b>3.1</b> Project information</a></li>
<li class="chapter" data-level="3.2" data-path="background-information.html"><a href="background-information.html#data-collection"><i class="fa fa-check"></i><b>3.2</b> Data collection</a></li>
<li class="chapter" data-level="3.3" data-path="background-information.html"><a href="background-information.html#data-availability"><i class="fa fa-check"></i><b>3.3</b> Data availability</a></li>
<li class="chapter" data-level="3.4" data-path="background-information.html"><a href="background-information.html#goal-of-the-exercise"><i class="fa fa-check"></i><b>3.4</b> Goal of the exercise</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="introduction-to-anvio.html"><a href="introduction-to-anvio.html"><i class="fa fa-check"></i><b>4</b> Introduction to Anvio</a><ul>
<li class="chapter" data-level="4.1" data-path="introduction-to-anvio.html"><a href="introduction-to-anvio.html#what-is-anvio"><i class="fa fa-check"></i><b>4.1</b> What is Anvi’o</a></li>
<li class="chapter" data-level="4.2" data-path="introduction-to-anvio.html"><a href="introduction-to-anvio.html#anvio-installation"><i class="fa fa-check"></i><b>4.2</b> Anvi’o installation</a></li>
<li class="chapter" data-level="4.3" data-path="introduction-to-anvio.html"><a href="introduction-to-anvio.html#binning-tools-installation"><i class="fa fa-check"></i><b>4.3</b> Binning tools installation</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="anvio-binning-exercise.html"><a href="anvio-binning-exercise.html"><i class="fa fa-check"></i><b>5</b> Anvio binning exercise</a><ul>
<li class="chapter" data-level="5.1" data-path="anvio-binning-exercise.html"><a href="anvio-binning-exercise.html#automatic-binning-on-anvio-using-metabat2-maxbin2-and-concoct"><i class="fa fa-check"></i><b>5.1</b> Automatic binning on Anvi’o using METABAT2, MAXBIN2, and CONCOCT</a><ul>
<li class="chapter" data-level="5.1.1" data-path="anvio-binning-exercise.html"><a href="anvio-binning-exercise.html#note-to-educe-tas"><i class="fa fa-check"></i><b>5.1.1</b> Note to EDUCE-TA’s</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="anvio-binning-exercise.html"><a href="anvio-binning-exercise.html#assessing-quality-bins"><i class="fa fa-check"></i><b>5.2</b> Assessing quality bins</a><ul>
<li class="chapter" data-level="5.2.1" data-path="anvio-binning-exercise.html"><a href="anvio-binning-exercise.html#note-the-mag-quality"><i class="fa fa-check"></i><b>5.2.1</b> Note the MAG quality</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="anvio-binning-exercise.html"><a href="anvio-binning-exercise.html#manual-refinement"><i class="fa fa-check"></i><b>5.3</b> Manual refinement</a></li>
<li class="chapter" data-level="5.4" data-path="anvio-binning-exercise.html"><a href="anvio-binning-exercise.html#iterative-assessment-and-refinement"><i class="fa fa-check"></i><b>5.4</b> Iterative assessment and refinement</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="processing-mags.html"><a href="processing-mags.html"><i class="fa fa-check"></i><b>6</b> Processing MAGs</a><ul>
<li class="chapter" data-level="6.1" data-path="processing-mags.html"><a href="processing-mags.html#taxonomic-classification-using-gtdb-tk"><i class="fa fa-check"></i><b>6.1</b> Taxonomic classification using GTDB-Tk</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i><b>7</b> References</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Introduction</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="anvio-binning-exercise" class="section level1">
<h1><span class="header-section-number">5</span> Anvio binning exercise</h1>
<p>You will be provided with contigs and merged profiles’ database to run binning and later manual refinement on the bins.</p>
<p>Please see this tutorial on <a href="http://merenlab.org/tutorials/assembly-based-metagenomics/">assembly-based metagenomics</a> and <a href="http://merenlab.org/2016/06/22/anvio-tutorial-v2/">metagenomic workflow tutorial</a> on Anvi’o to understand how these files were obtained and an in-depth explanation on moving forward. The Anvi’o documentation also provides an excellent commentary on the nuances of each step.</p>
<div id="automatic-binning-on-anvio-using-metabat2-maxbin2-and-concoct" class="section level2">
<h2><span class="header-section-number">5.1</span> Automatic binning on Anvi’o using METABAT2, MAXBIN2, and CONCOCT</h2>
<p>The following commands provide a step by step approach to obtain bins and to load the results on the visualization platform of Anvi’o on your browser for manual refinement.</p>
<pre><code>#Create variables for profile and contig databases for easy access to files
profile_db = ~/path_to_profile_db 
contig_db = ~/path_to_contig_db 

#Make output directory
Mkdir ~/path_to_output_directory

#Create variable for output directory
bin_out=~/path_to_output_directory

#Operational parameters
threads=32 #Change to suitable amount of threads

#Clustering with metabat2
anvi-cluster-contigs -p $profile_db -c $contig_db -C metabat --driver metabat2 -T $threads --just-do-it

#Clustering with concoct
anvi-cluster-contigs -p $profile_db -c $contig_db -C concoct --driver concoct -T $threads --just-do-it

#Clustering with maxbin2
anvi-cluster-contigs -p $profile_db -c $contig_db -C maxbin --driver maxbin2 -T $threads --just-do-it

#Consensus binning with DASTOOL
anvi-cluster-contigs -p $profile_db -c $contig_db --driver dastool -S metabat,concoct,maxbin -C dastool --just-do-it -T $threads

#Obtain bins as fasta files
anvi-summarize -p $profile_db -c $contig_db -C dastool -o $bin_out/dastool_out

#Copy all bin fasta files for downstream processing such as quality estimation, taxonomic classification, and functional annotation

#Make directory for fasta files for easy access
mkdir $bin_out/fasta_files
cp $bin_out/dastool_out/*/*fasta $bin_out/fasta_files</code></pre>
<div id="note-to-educe-tas" class="section level3">
<h3><span class="header-section-number">5.1.1</span> Note to EDUCE-TA’s</h3>
<p>In case we want the students to practice manual refinement alone, we can provide them with contaminated bins using Concoct.</p>
<p>The following commands creates a fixed number of bins (ideally lower than the actual genomes expected) using Concoct</p>
<pre><code>profile_db = ~/path_to_profile_db 
contig_db = ~/path_to_contig_db 

#Make output directory
Mkdir ~/path_to_output_directory

#Operational parameters
threads=32 #Change to suitable amount of threads

bins=N #Change N to half of the number of bins expected to deliberately contaminate bins and later manually refine them

#Clustering with concoct
anvi-cluster-contigs -p $profile_db -c $contig_db -C concoct --driver concoct -T $threads --just-do-it --clusters $bins</code></pre>
</div>
</div>
<div id="assessing-quality-bins" class="section level2">
<h2><span class="header-section-number">5.2</span> Assessing quality bins</h2>
<p>Check completion and redundancy of MAGs using CheckM. Change paths and operational parameters as required for your system. For a more detailed understanding of how to use CheckM please visit <a href="https://github.com/Ecogenomics/CheckM/wiki">CheckM wiki</a></p>
<pre><code>#Create checkM output directory
mkdir ~/path_to_checkm_output_dir

#Create variable for output directory
checkm_out=~/path_to_checkm_output_dir

#Establish bins’ fasta files directory 
in_fasta=~/path_to_output_fasta_files

#Operational parameters – change as required on your system
threads=48
pplacer_threads=24

#Run CheckM
checkm lineage_wf -t $threads --pplacer_threads $pplacer_threads \
    -f $checkm_out/checkm_out.tsv --tab_table \
    -x fasta $in_fasta $checkm_out/checkm_lineage_wf
    
#checkm_out.tsv is a tab separated file that provides quality estimates on the set of bins</code></pre>
<div id="note-the-mag-quality" class="section level3">
<h3><span class="header-section-number">5.2.1</span> Note the MAG quality</h3>
<p>Open the tab separated file, checkm_out.tsv in a spreadsheet viewer and study the completion and contamination column. We are looking for high/medium quality MAGs For a detailed commentary on high-quality MAGs, please refer <a href="https://rdcu.be/b9JqO">Bowers et al. 2017</a></p>
<p>For this exercise, we aim to have bins with less than 10 % contamination and more than 50 % completion. We would now use the “anvi-refine” feature to manually refine the bins which have more than 10 % contamination. Please make a note of all the bins which are more than 50 % complete and have more than 10 % contamination.</p>
</div>
</div>
<div id="manual-refinement" class="section level2">
<h2><span class="header-section-number">5.3</span> Manual refinement</h2>
<p>Manual refinement on Anvi’o is described in great detail <a href="http://merenlab.org/2017/05/11/anvi-refine-by-veronika/">here</a></p>
<iframe src="http://merenlab.org/2017/05/11/anvi-refine-by-veronika/?showcase=0" width="672" height="600px">
</iframe>
</div>
<div id="iterative-assessment-and-refinement" class="section level2">
<h2><span class="header-section-number">5.4</span> Iterative assessment and refinement</h2>
<p>After manual refinement of the bins on Anvi’o, iteratively run quality assessment and manual refinement of high contamination bins until you have a list of bins which are more than 90 % complete and less than 10 % contaminated.</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="introduction-to-anvio.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="processing-mags.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
