<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>2 Instructions | Anvi’o binning and refinement tutorial</title>
  <meta name="description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="2 Instructions | Anvi’o binning and refinement tutorial" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="2 Instructions | Anvi’o binning and refinement tutorial" />
  
  <meta name="twitter:description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  

<meta name="author" content="Pranav_Sampara" />


<meta name="date" content="2020-05-11" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="index.html"/>
<link rel="next" href="background-information.html"/>
<script src="libs/header-attrs-2.5/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<link href="libs/anchor-sections-1.0/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0/anchor-sections.js"></script>



<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Anvio binning and refinement tutorial</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#guide-to-the-book"><i class="fa fa-check"></i><b>1.1</b> Guide to the book</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#brief-introduction-to-omics-methods"><i class="fa fa-check"></i><b>1.2</b> Brief introduction to ’omics methods</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#some-key-aspects-of-metagenomics"><i class="fa fa-check"></i><b>1.3</b> Some key aspects of metagenomics</a></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#key-points-to-be-aware-of-and-known-pitfalls-in-metagenomics"><i class="fa fa-check"></i><b>1.4</b> Key points to be aware of and known pitfalls in metagenomics</a></li>
<li class="chapter" data-level="1.5" data-path="index.html"><a href="index.html#basic-outline-of-a-metagenomics-pipeline"><i class="fa fa-check"></i><b>1.5</b> Basic outline of a metagenomics pipeline</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="instructions.html"><a href="instructions.html"><i class="fa fa-check"></i><b>2</b> Instructions</a>
<ul>
<li class="chapter" data-level="2.1" data-path="instructions.html"><a href="instructions.html#goal-of-the-exercise"><i class="fa fa-check"></i><b>2.1</b> Goal of the exercise</a></li>
<li class="chapter" data-level="2.2" data-path="instructions.html"><a href="instructions.html#project-ideas"><i class="fa fa-check"></i><b>2.2</b> Project ideas</a></li>
<li class="chapter" data-level="2.3" data-path="instructions.html"><a href="instructions.html#timeline"><i class="fa fa-check"></i><b>2.3</b> Timeline</a></li>
<li class="chapter" data-level="2.4" data-path="instructions.html"><a href="instructions.html#reports"><i class="fa fa-check"></i><b>2.4</b> Reports</a>
<ul>
<li class="chapter" data-level="2.4.1" data-path="instructions.html"><a href="instructions.html#abstract"><i class="fa fa-check"></i><b>2.4.1</b> Abstract</a></li>
<li class="chapter" data-level="2.4.2" data-path="instructions.html"><a href="instructions.html#introduction-1"><i class="fa fa-check"></i><b>2.4.2</b> Introduction</a></li>
<li class="chapter" data-level="2.4.3" data-path="instructions.html"><a href="instructions.html#methods"><i class="fa fa-check"></i><b>2.4.3</b> Methods</a></li>
<li class="chapter" data-level="2.4.4" data-path="instructions.html"><a href="instructions.html#research-questions"><i class="fa fa-check"></i><b>2.4.4</b> Results</a></li>
<li class="chapter" data-level="2.4.5" data-path="instructions.html"><a href="instructions.html#discussion"><i class="fa fa-check"></i><b>2.4.5</b> Discussion</a></li>
<li class="chapter" data-level="2.4.6" data-path="instructions.html"><a href="instructions.html#references"><i class="fa fa-check"></i><b>2.4.6</b> References</a></li>
</ul></li>
<li class="chapter" data-level="2.5" data-path="instructions.html"><a href="instructions.html#assessment"><i class="fa fa-check"></i><b>2.5</b> Assessment</a>
<ul>
<li class="chapter" data-level="2.5.1" data-path="instructions.html"><a href="instructions.html#peer-evaluation-5"><i class="fa fa-check"></i><b>2.5.1</b> Peer evaluation (5%)</a></li>
<li class="chapter" data-level="2.5.2" data-path="instructions.html"><a href="instructions.html#report-10"><i class="fa fa-check"></i><b>2.5.2</b> Report (10%)</a></li>
</ul></li>
<li class="chapter" data-level="2.6" data-path="instructions.html"><a href="instructions.html#ujemi-submission"><i class="fa fa-check"></i><b>2.6</b> UJEMI submission</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="background-information.html"><a href="background-information.html"><i class="fa fa-check"></i><b>3</b> Background information</a>
<ul>
<li class="chapter" data-level="3.1" data-path="background-information.html"><a href="background-information.html#project-information"><i class="fa fa-check"></i><b>3.1</b> Project information</a></li>
<li class="chapter" data-level="3.2" data-path="background-information.html"><a href="background-information.html#data-collection"><i class="fa fa-check"></i><b>3.2</b> Data collection</a></li>
<li class="chapter" data-level="3.3" data-path="background-information.html"><a href="background-information.html#data-availability"><i class="fa fa-check"></i><b>3.3</b> Data availability</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="introduction-to-anvio.html"><a href="introduction-to-anvio.html"><i class="fa fa-check"></i><b>4</b> Introduction to Anvi’o</a>
<ul>
<li class="chapter" data-level="4.1" data-path="introduction-to-anvio.html"><a href="introduction-to-anvio.html#what-is-anvio"><i class="fa fa-check"></i><b>4.1</b> What is Anvi’o</a></li>
<li class="chapter" data-level="4.2" data-path="introduction-to-anvio.html"><a href="introduction-to-anvio.html#anvio-installation"><i class="fa fa-check"></i><b>4.2</b> Anvi’o installation</a></li>
<li class="chapter" data-level="4.3" data-path="introduction-to-anvio.html"><a href="introduction-to-anvio.html#binning-tools-installation"><i class="fa fa-check"></i><b>4.3</b> Binning tools installation</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="anvio-binning-exercise.html"><a href="anvio-binning-exercise.html"><i class="fa fa-check"></i><b>5</b> Anvi’o binning exercise</a>
<ul>
<li class="chapter" data-level="5.1" data-path="anvio-binning-exercise.html"><a href="anvio-binning-exercise.html#automatic-binning-on-anvio-using-metabat2-maxbin2-and-concoct"><i class="fa fa-check"></i><b>5.1</b> Automatic binning on Anvi’o using METABAT2, MAXBIN2, and CONCOCT</a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="anvio-binning-exercise.html"><a href="anvio-binning-exercise.html#note-to-educe-tas"><i class="fa fa-check"></i><b>5.1.1</b> Note to EDUCE-TA’s</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="anvio-binning-exercise.html"><a href="anvio-binning-exercise.html#assessing-quality-bins"><i class="fa fa-check"></i><b>5.2</b> Assessing quality bins</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="anvio-binning-exercise.html"><a href="anvio-binning-exercise.html#note-the-mag-quality"><i class="fa fa-check"></i><b>5.2.1</b> Note the MAG quality</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="anvio-binning-exercise.html"><a href="anvio-binning-exercise.html#manual-refinement"><i class="fa fa-check"></i><b>5.3</b> Manual refinement</a></li>
<li class="chapter" data-level="5.4" data-path="anvio-binning-exercise.html"><a href="anvio-binning-exercise.html#iterative-assessment-and-refinement"><i class="fa fa-check"></i><b>5.4</b> Iterative assessment and refinement</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="processing-mags.html"><a href="processing-mags.html"><i class="fa fa-check"></i><b>6</b> Processing MAGs</a>
<ul>
<li class="chapter" data-level="6.1" data-path="processing-mags.html"><a href="processing-mags.html#taxonomic-classification-using-gtdb-tk"><i class="fa fa-check"></i><b>6.1</b> Taxonomic classification using GTDB-Tk</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="references-1.html"><a href="references-1.html"><i class="fa fa-check"></i><b>7</b> References</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Anvi’o binning and refinement tutorial</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="instructions" class="section level1" number="2">
<h1><span class="header-section-number">2</span> Instructions</h1>
<div id="goal-of-the-exercise" class="section level2" number="2.1">
<h2><span class="header-section-number">2.1</span> Goal of the exercise</h2>
<p>The goal is to generate high quality metagenome assembled genomes (MAGs), identify the taxonomy, and the metabolic potential of the MAGs of interest.</p>
<p>You will be working with metagenomic data from Cruise #. You will be provided with prerequisite information, such as a merged profile database and contigs database, that you would use to bin MAGs and later classify the taxonomy. You would be working on an ’omics analysis and visualization platform, Anvi’o. For binning, you would use Metabat2, Maxbin2, Concoct, and DASTool. For taxonomic classification you would use CheckM and finally to check genome completeness and redundancy you would use CheckM.</p>
</div>
<div id="project-ideas" class="section level2" number="2.2">
<h2><span class="header-section-number">2.2</span> Project ideas</h2>
<ul>
<li>Determine what fractions of contigs map to MAGs vs contigs not in MAGs, as a way to evaluate the total amount of information available to evaluate</li>
<li>Evaluate nitrogen cycling genes within MAGs using Prokka and TreeSAPP annotations -Compare taxonomic assignment TreeSAPP identified genes to GTDB-Tk assigned taxonomies of MAGs, generated using Anvi’o. Can you identify known groups based on published work -Evaluate changes in nitrogen cycling gene taxonomy and abundance at different depths in the Saanich inlet water column</li>
</ul>
</div>
<div id="timeline" class="section level2" number="2.3">
<h2><span class="header-section-number">2.3</span> Timeline</h2>
<p>The following provides an outline as well as some specific milestones within the project.</p>
</div>
<div id="reports" class="section level2" number="2.4">
<h2><span class="header-section-number">2.4</span> Reports</h2>
<p>Reports should be formatted as per the <a href="https://jb.asm.org/sites/default/files/additional-assets/JB-ITA.pdf">Instructions to Authors</a> for the <a href="https://jb.asm.org/">Journal of Bacteriology</a>.</p>
<p>Each group will complete <strong>one</strong> report with the following sections.</p>
<div id="abstract" class="section level3" number="2.4.1">
<h3><span class="header-section-number">2.4.1</span> Abstract</h3>
<p><em>200–250 words</em></p>
<p>Note that an Importance section is not required.</p>
</div>
<div id="introduction-1" class="section level3" number="2.4.2">
<h3><span class="header-section-number">2.4.2</span> Introduction</h3>
<p><em>500–750 words</em></p>
<ul>
<li>Overview of the nitrogen cycle including its global impacts and microbial foundations.</li>
<li>Introduce Saanich Inlet as a model ecosystem for studying microbial community responses to ocean deoxygenation <em>e.g.</em> seasonal cycles, relevant biogeochemistry, previous studies, etc.</li>
</ul>
</div>
<div id="methods" class="section level3" number="2.4.3">
<h3><span class="header-section-number">2.4.3</span> Methods</h3>
<p><em>300–500 words</em></p>
<ul>
<li><p>Briefly describe the data (sampling, sequencing, processing, etc.)</p></li>
<li><p>Briefly describe your analysis methods including</p>
<ul>
<li>Anvi’o version and commands used</li>
<li>Binning tools version and commands used</li>
<li>CheckM and GTDB-Tk version and commands used. For GTDB-Tk, please mention the data release that was used</li>
<li>R version and packages used</li>
<li>Statistics (if applicable)</li>
</ul></li>
<li><p>Provide three shell scripts and one single R script (if any) (i.e “binning.sh”, “completion_redundancy.sh”, “taxonomic_classification.sh” and an R script “file_name.R”) as individual files (i.e. not as part of your manuscript) containing the complete code to generate your results.</p></li>
</ul>
</div>
<div id="research-questions" class="section level3" number="2.4.4">
<h3><span class="header-section-number">2.4.4</span> Results</h3>
<p><em>500–750 words</em></p>
<p>Your analysis will focus on <strong>binning and manual refinement, generating high quality bins (assessed by CheckM), assigning taxonomy, and metabolic networks of interested taxa and gene modules (typically denitrification genes, but you can be creative)</strong> (which is given in your Canvas group name) and the following questions:</p>
<p>You must include ≥ figures/panels with titles and full captions. These figures can be combined into multi-panel figures if desired. These figures would typically contain - The Anvi’o metagenome visualization and an example of a refined bin. - Completion and redundancy estimates with GTDB-Tk taxonomic classification at phylum and family levels - Plot depth of MAG coverage relative to the number of reads in the metagenome for each depth - Generate annotations for predicted ORFs on contigs and MAGs using Prokka, and - Compare recovery of nitrogen cycling genesbetween MAGs and assembled contigs between Prokka and TreeSAPP</p>
</div>
<div id="discussion" class="section level3" number="2.4.5">
<h3><span class="header-section-number">2.4.5</span> Discussion</h3>
<p><em>750–1000 words</em></p>
<ul>
<li>Frame your depth’s results within a broader discussion of Saanich Inlet and the other depths (Apr 6 discussion)</li>
<li>Propose expected taxa and comment on coverage, and the reasoning behind these results<br />
</li>
<li>Provide a discussion on metabolic networks of the interested taxa</li>
<li>Future directions</li>
</ul>
</div>
<div id="references" class="section level3" number="2.4.6">
<h3><span class="header-section-number">2.4.6</span> References</h3>
<p><em>10 or more</em> formatted in the ASM style such as for the <a href="https://jb.asm.org/content/organization-and-format">Journal of Bacteriology</a>. If you are using a reference manager, this style can be downloaded for <a href="https://endnote.com/style_download/american-society-for-microbiology-asm-journals-2/">EndNote</a>, <a href="https://github.com/citation-style-language/styles/blob/master/american-society-for-microbiology.csl">Mendeley</a>, or <a href="https://www.zotero.org/styles?q=microbiology">Zotero</a>.</p>
<p>Make sure to cite the data source papers!</p>
</div>
</div>
<div id="assessment" class="section level2" number="2.5">
<h2><span class="header-section-number">2.5</span> Assessment</h2>
<p>Reports will account for 15% of the course marks distributed as follows.</p>
<div id="peer-evaluation-5" class="section level3" number="2.5.1">
<h3><span class="header-section-number">2.5.1</span> Peer evaluation (5%)</h3>
<p><strong>Total: 40 pts</strong></p>
<p>Peer assessment will occur through <a href="https://www.catme.org/">CATME</a> wherein everyone will assess themselves and their group members. Marks will be scaled to your evaluation score with 90%+ (or 0.9+ out of 1 within CATME) counting as full marks.</p>
<p>Individuals who do not adequately contribute to their group will forfeit the 5% contributions mark and may be asked to complete the project on their own.</p>
<p>Peer assessment includes:</p>
<ul>
<li>Contributing to the team’s work</li>
<li>Interacting with teammates</li>
<li>Expecting quality</li>
<li>Having related knowledge, skills, and abilities</li>
</ul>
<p>Individuals score themselves and all team members on a scale of 1 (unsatisfactory) – 3 (satisfactory) – 5 (excellent) in these categories. Scores are then averaged and scaled within team such that 1 corresponds to all team members contributing equally to all categories of assessment:</p>
<ul>
<li>1+ = Mostly 5s</li>
<li>0.95 = Mostly 4s</li>
<li>0.8 = Mostly 3s</li>
<li>0.65 = Mostly 2s</li>
<li>0.5 = Mostly 1s</li>
</ul>
<p>Marks will be scaled to your CATME score based on the following:</p>
<ul>
<li>0.9 - 1+ = 100% = 40</li>
<li>0.8 - 0.89 = 90% = 36</li>
<li>0.7 - 0.79 = 80% = 32</li>
<li>0.6 - 0.69 = 70% = 28</li>
<li>&lt;0.6 = Assessed on a case-by-case basis</li>
</ul>
<p><em>Unwarranted blanketed low scoring of your team or high scoring of yourself will result in the removal of your scores from the average so that they do not unfairly impact you or your team members.</em></p>
<p>If issues occur within your group, please do not hesitate to contact Dr. Hallam (<a href="mailto:shallam@mail.ubc.ca" class="email">shallam@mail.ubc.ca</a>), or Dr. Koenig (<a href="mailto:stephan.koenig@ubc.ca" class="email">stephan.koenig@ubc.ca</a>) <em>at any point</em>.</p>
</div>
<div id="report-10" class="section level3" number="2.5.2">
<h3><span class="header-section-number">2.5.2</span> Report (10%)</h3>
<p><strong>Total: 80 pts</strong></p>
<p>Each group will submit an electronic copy of their group’s final report (due <strong>April 24</strong>). Reports will be assessed on:</p>
<ul>
<li>Completion of relevant analyses toward answering biological questions</li>
<li>Logic and completeness of conclusions made from these analyses</li>
<li>Writing clarity, grammar, and style</li>
<li>Figure clarity, effectiveness, and relevance</li>
</ul>
<table>
<colgroup>
<col width="3%" />
<col width="19%" />
<col width="23%" />
<col width="23%" />
<col width="29%" />
</colgroup>
<thead>
<tr class="header">
<th></th>
<th><strong>Poor (0-1 pts)</strong></th>
<th><strong>Below average (2-4 pts)</strong></th>
<th><strong>Good (5-7 pts)</strong></th>
<th><strong>Excellent (8-10 pts)</strong></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><strong>Abstract</strong></td>
<td>Not provided</td>
<td>Summarizes only part of the report or contains numerous inaccuracies</td>
<td>Summarizes most of the report including relevant results; Some inaccuracies or missing pieces</td>
<td>Accurately summarizes the report including background, relevant results, and conclusions; Few to no inaccuracies</td>
</tr>
<tr class="even">
<td><strong>Introduction</strong></td>
<td>Background information not provided or provided but not relevant to the research question(s); Hypotheses / questions are not stated nor supported by relevant sources</td>
<td>Background is relevant but not sufficient to frame the research question(s); Hypotheses / questions are unclear and/or unsupported</td>
<td>Background is relevant but does not fully frame the research question(s); Hypotheses / questions are unclear or unsupported</td>
<td>Background is relevant and fully frames the research question(s); Hypotheses / questions are clearly stated and supported by scientific sources</td>
</tr>
<tr class="odd">
<td><strong>Methods</strong></td>
<td>Incomplete and missing numerous methods used; Unclear or confusing</td>
<td>Missing several methods used; Numerous errors or unclear statements</td>
<td>Missing one or more methods used; Results could not be replicated due to issues with clarity or accuracy</td>
<td>Sufficient to allow replication of the results including sampling, sequencing, processing, and analysis; Succinct and not overly wordy</td>
</tr>
<tr class="even">
<td><strong>Code (R and TreeSAPP)</strong></td>
<td>Code missing significant portions or is not provided</td>
<td>Code is incomplete or erroneous such that results cannot be replicated; No explanation of code is given</td>
<td>Code replicates results but is verbose or inefficient; Minimal commenting is provided</td>
<td>Code replicates results and is efficient; Detailed explanation of code is provided in comments using # notation</td>
</tr>
<tr class="odd">
<td><strong>Results</strong></td>
<td>Most of the research questions are not addressed or the results are consistently incomplete or irrelevant</td>
<td>Several research questions are not addressed; Numerous results are incomplete, erroneous, or irrelevant</td>
<td>1 or more research questions are not fully addressed; Some results are incomplete or do not to pertain to the questions</td>
<td>All research questions are investigated with relevant analyses and figures; Results are clearly stated and not incorrectly or over-interpreted</td>
</tr>
<tr class="even">
<td><strong>Figures &amp; captions</strong></td>
<td>2 or fewer figures are given; Captions are incomplete or missing</td>
<td>3+ relevant figures are provided but lack proper formatting or completeness; Captions lack numerous details or descriptions</td>
<td>4+ relevant figures are provided with mostly proper formatting; Captions lack some details or descriptions</td>
<td>5+ relevant figures are provided with proper formatting; Captions are included for all figures and contain a title as well as description of axes, other aesthetics, and overall data trends</td>
</tr>
<tr class="odd">
<td><strong>Discussion</strong></td>
<td>Synthesis of results is missing or does not pertain to the data; Statements are not supported by the data</td>
<td>Conclusions are incomplete or not relevant to the data; Multiple research questions are not addressed or results are incorrectly interpreted</td>
<td>Conclusions restate results but without summarization or synthesis across depths; 1 or more questions are not addressed; Some results are incorrectly interpreted</td>
<td>Results are summarized within the context of the original questions and more broadly across depths; Results are correctly interpreted and discussed with scientific language; Conclusions are supported by the data and 1 or more future directions are proposed</td>
</tr>
<tr class="even">
<td><strong>Writing, grammar, &amp; format</strong></td>
<td>Grammatical errors, spelling mistakes, and/or language cause significant issues in understanding of the content throughout</td>
<td>Numerous grammatical, spelling, or language errors that negatively impact understanding; Some language is overly verbose or informal; One or more sections significantly differ from the recommended length</td>
<td>Some grammatical, spelling, or language errors that negatively impact understanding occur; Some language is overly verbose or informal; One section may significantly differ from the recommended length</td>
<td>Minimal to no errors; Formal scientific language used throughout; Text flows and is easy to read; Sections adhere to word limits; Report provided in J Bacteriology format</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="ujemi-submission" class="section level2" number="2.6">
<h2><span class="header-section-number">2.6</span> UJEMI submission</h2>
<p>Outstanding reports will be invited to submit to The Undergraduate Journal of Experimental Microbiology and Immunology (<a href="https://jemi.microbiology.ubc.ca/">UJEMI</a>). More information on this will be provided near the end of the Finals Period.</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="index.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="background-information.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "section"
},
"link": "GitHub repo link"
});
});
</script>

</body>

</html>
